### ISSUE TITLE
High i/o write bandwidth when writing WAL

### ISSUE THREAD

**Issue by manio** [influxdata/influxdb#25128]
Hi,
I have influxdb v1.8.10-1 from `repos.influxdata.com/debian`.
I have several databases which are mostly set for 6 months retention policy.
I noticed that the WAL saving is making quite a lot of write I/O into the `/var/lib/influxdb/wal`.
As this WAL dir was roughly 100MB in my case, I copied it to the `/dev/shm`, then bind-mounted it there:
`mount -o bind /dev/shm/wal /var/lib/influxdb/wal`

And take a look at the following pic:
![image](https://github.com/influxdata/influxdb/assets/583157/e81e786c-688b-451d-ada7-5aa8d1b0754b)
I have spinning HDD and it was making a terrible noise writing this WAL, after the change it is almost silent.

Why is this WAL saving so much I/O intensive?
Can it be adjusted somehow to not commit as much "aggressively"?

**Comment by davidby-influx** [comment]
https://docs.influxdata.com/influxdb/v1/administration/config/#wal-fsync-delay--0s

**Comment by manio** [comment]
Setting this value only makes a writing resolution worse:
![image](https://github.com/influxdata/influxdb/assets/583157/f95192d8-89ee-4072-82c5-39a7d38e519b)

**Comment by davidby-influx** [comment]
@manio - I am unsure what your voltage graph signifies. Changing the `wal-fsync-delay` batches together writes to the WAL before they are committed to the disk with fsync, but does not change the data written in any way.  

Queries do not use the WAL; when a write is received by InfluxDB, it is stored in an in-memory cache and written to disk in the WAL. In normal operation, the cache is flushed to the disk as TSM files; the WAL is only read if the system stops before the cache is flushed; in this case, on startup the WAL will be read into the cache

Here is a better description of how the WAL is used:

https://docs.influxdata.com/influxdb/v2/reference/internals/storage-engine/#write-ahead-log-wal

In short, whatever effect you are seeing in the data is unrelated to the wal-fsync-delay, and the only purpose of the wal-fsync-delay is to fsync the WAL less frequently and reduce disk load.

**Comment by manio** [comment]
@davidby-influx 
Thank you for your lengthy explanation, however the `wal-fsync-delay` was the only single parameter I changed and I can clearly see a difference in a resolution of resulting graph. This data is written every 5 seconds.

Take a look at the opposite, when I reverted the changes:
![image](https://github.com/influxdata/influxdb/assets/583157/71ec9a54-7a85-4aca-b246-243101855036)
or eg here:
![image](https://github.com/influxdata/influxdb/assets/583157/af08d4d3-e2d4-4fe8-865f-cfe1a59f7ea2)

I didn't mention though, that the 100ms recommenced max value was not enough and I've changed it to 5 sec. And maybe this was the main problem.
Finally I sticked with the `wal` in the `/dev/shm` for now, because I don't have a dedicated server room (I am using it in my homelab) and cannot stand the HDD noise ;)

**Comment by davidby-influx** [comment]
It would be interesting to know if you are actually getting fewer points written with high `wal-fsync-delay` or if this is merely a graphic artifact. I do not know of any reason why change the fsync delay would affect the data received (unless you are dropping points because of system overload or some other error).  No need to investigate further, though, if you are satisfied with the situation as it is now.
